src/Algebra/HeytingField.agda

```diff
@@ -0,0 +1,99 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Local commutative rings and heyting fields
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Algebra.HeytingField where
+
+open import Algebra.Core using (Op₁; Op₂)
+open import Algebra.Structures using (IsCommutativeRing)
+open import Relation.Binary.Core using (Rel)
+open import Relation.Binary.Structures using (IsEquivalence)
+open import Level using (Level; _⊔_) renaming (suc to lsuc)
+open import Data.Product using (∃-syntax; _×_)
+
+import Relation.Binary.Apartness as A
+
+
+module Structures
+  {c ℓ₁ ℓ₂} {Carrier : Set c}
+  (_≈_ : Rel Carrier ℓ₁)
+  (_#_ : Rel Carrier ℓ₂)
+  (_+_ _*_ : Op₂ Carrier) (-_ : Op₁ Carrier) (0# 1# : Carrier)
+  where
+
+  open A.Structures hiding (¬#-equiv)
+
+  -- does this need to be public?
+  private
+    invertible : Carrier → Set _
+    invertible x = ∃[ 1/x ] (x * 1/x) ≈ 1#
+
+
+  record IsLocalCommutativeRing : Set (c ⊔ ℓ₁ ⊔ ℓ₂) where
```

Comment:
In contrast to apartness relations, it seems to me there probably is a deep hierarchy of "local" structures that use the apartness relation, so maybe they should live in their own hierarchy. What do you think about them living under `Algebra.Local`? Open to better naming suggestions!

Just like the main algebra hierarchy they would once again have separate modules e.g. `Algebra.Local.Structures`/`Algebra.Local.Bundles` etc.

Reply:
I've done this, but I won't resolve this thread as it's not obvious what the best route is.

Reply:
ok I've put this in Algebra.WithApartness now. See zulip discussions (I'll bring it up again there before resolving this thread).

Reply:
I think we've reached consensus on `Algebra.Apartness` in the thread.